word_search_puzzle.py

```diff
@@ -1,2 +1,117 @@
+import random
+
 print("Welcome to the Word Search Puzzle!")
-print("Let's start!")
\ No newline at end of file
+print("Let's start!")
+
+def is_space_available(word, empty_board, row, col, direction):  
+    if direction == 0: 
+        if col + len(word) > len(empty_board[0]): 
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row][col + l] not in ["_", word[l]]:
+                    return False
+    
+    elif direction == 1:  
+        if row + len(word) > len(empty_board): 
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row + l][col] not in ["_", word[l]]:
+                    return False
+    
+    elif direction == 2:  
+        if row + len(word) > len(empty_board) or col + len(word) > len(empty_board[0]):
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row + l][col + l] not in ["_", word[l]]:
+                    return False
+    
+    else:  
+        if row - len(word) < -1 or col - len(word) < -1:
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row - l][col - l] not in ["_", word[l]]:
+                    return False
+    
+    return True
+
+def place_word(word, empty_board):
+    placed = False
+    max_attempts = 100
+    attempts = 0
+    while not placed and attempts < max_attempts:
+        reverse = random.randint(0, 1)
+        if reverse:
+            word = word[::-1]
+
+        pos = random.randint(0, 3)  # 0 horizontal, 1 vertical, 2 diag_right, 3 diag_left
```

Comment:
Avoid [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)). Here we could use an `enum`. One way of using it would be like [this](https://stackoverflow.com/questions/24243500/random-choice-on-enum).